Client/Frontend/Browser/Playlist/VideoPlayer/UI/VideoPlayer.swift

```diff
@@ -202,9 +223,160 @@ class VideoView: UIView, VideoTrackerBarDelegate {
+        if event?.type == UIEvent.EventType.touches {
+            dragStartedVolume = AVAudioSession.sharedInstance().outputVolume
+
+            toggleOverlays(showOverlay: true)
+            if let currentPlayer = playerLayer?.player {
+                let currentSeconds = CGFloat(currentPlayer.currentTime().value) / CGFloat((currentPlayer.currentTime().timescale))
+                dragStartedTimelinePos = currentSeconds
+            }
+        }
+
+        super.touchesBegan(touches, with: event)
+    }
+
+    override func touchesEnded(_ touches: Set<UITouch>, with event: UIEvent?) {
+        if event?.type == UIEvent.EventType.touches {
+            toggleOverlays(showOverlay: false)
+        }
+
+        super.touchesEnded(touches, with: event)
     }
     
+    @objc
+    private func onOverlayDragged(_ gestureRecognizer: UIPanGestureRecognizer) {
+        if gestureRecognizer.state == .ended {
+            // User ended panning
+            currentDraggingDirection = .noDirection
+            
+            guard let delegate = delegate else { return }
+
+            isSeeking = false
+
+            if wasPlayingBeforeSeeking {
+                delegate.play(self)
+                delegate.setPlaybackRate(self, rate: playbackRate)
+                wasPlayingBeforeSeeking = false
+            }
+
+            if delegate.isPlaying {
+                fadeAnimationWorkItem?.cancel()
+                toggleOverlays(showOverlay: false, except: [overlayView], display: [overlayView])
+                overlayView.alpha = 0.0
+                isOverlayDisplayed = false
+            } else {
+                fadeAnimationWorkItem?.cancel()
+                toggleOverlays(showOverlay: true)
+                overlayView.alpha = 1.0
+                isOverlayDisplayed = true
+            }
+
+        } else {
+            if let currentPlayer = playerLayer?.player {
+
+                // Horizontal or vertical pan?
+                if currentDraggingDirection == .noDirection {
+                    if abs(gestureRecognizer.velocity(in: self).y) > abs(gestureRecognizer.velocity(in: self).x) {
+                        currentDraggingDirection = .verticalDirection
+                    } else {
+                        currentDraggingDirection = .horizontalDirection
+                    }
+                }
+
+                if currentDraggingDirection == .verticalDirection {
+
+                    // Vertical dragging adjusts volume.
+                    let vDragRatio = 4*gestureRecognizer.translation(in: self).y/self.bounds.height // (-0.5 ... 0.5)*4
+                    
+                    if vDragRatio != 0 && self.dragStartedVolume >= 0 {
+                        if let slider = volumeView.subviews.first(where: { $0 is UISlider }) as? UISlider {
+                            var finalVal: Float = 0
+                            finalVal = self.dragStartedVolume + Float(-vDragRatio)
```

Comment:
`let finalVal = self.dragStartedVolume -vDragRatio`

Reply:
I think here we do need explicit type conversion as one operand is CGFloat and the other one is Float.